Mathlib/Tactic/LinearOptim/Main.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2025. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Shuli Chen, Robert Y. Lewis, Heather Macbeth, Siqing Zhang, Runtian Zhou
+-/
+import Lean.Meta.Basic
+import Mathlib.Control.Basic
+import Mathlib.Data.Ineq
+import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Linarith.Preprocessing
+import Mathlib.Tactic.Linarith.Verification
+import Mathlib.Tactic.Linarith.Oracle.SimplexAlgorithm.PositiveVector
+import Mathlib.Tactic.Polyrith
+import Mathlib.Tactic.Ring.Basic
+import Mathlib.Util.Qq
+import Mathlib.Tactic.CancelDenoms.Core
+
+/-!
+# Linear Optimization Tactics
+
+This file implements `maximize` and `minimize` tactics that find upper and lower
+bounds for linear expressions given linear constraints using the simplex algorithm.
+
+## Main declarations
+
+* `maximize` - Finds an upper bound for a linear expression
+* `minimize` - Finds a lower bound for a linear expression
+* `findPositiveVector` - Finds a positive vector solution using simplex algorithm
+* `runSimplexAlgorithm` - Core simplex algorithm implementation
+
+## Implementation
+
+Both tactics work by:
+1. Parsing the linear constraints and target expression
+2. Converting them to matrix form suitable for the simplex algorithm
+3. Finding the optimal solution using simplex with Bland's rule for pivoting
+4. Suggesting a `have` statement with the computed bound
+
+The certificate search is reduced to finding a nonnegative vector `v` such that some
+coordinates from strict inequalities are positive and `A v = 0`. This is solved using:
+
+1. Translation to a Linear Programming problem
+2. Gaussian elimination to get initial tableau
+3. Simplex algorithm with Bland's rule until solution is found
+-/
+
+open Lean Lean.Elab Lean.Elab.Tactic Lean.Meta
+open Mathlib.Tactic.Linarith Mathlib.Tactic.Linarith.SimplexAlgorithm
+open Qq
+
+namespace Mathlib.Tactic.LinearOptim
+
+section Preprocessing
+
+/-- The default preprocessors for the linear optimization tactics.
+Throws away non-linear-inequality hypotheses, pushes negations, turns inequalities into ≤,
+moves terms to the left hand side, and cancels denominators. -/
+def defaultPreprocessors : List Preprocessor :=
+  [filterComparisons, removeNegations, strengthenStrictInt, compWithZero, cancelDenoms]
```

Comment:
Move this list to right after `Mathlib.Tactic.Linarith.defaultPreprocessors` (and rename, maybe to `defaultNonbranchingPreprocessors`)